std/system.d

```diff
@@ -82,5 +82,76 @@ immutable
     /// The endianness that the program was compiled for.
     version (LittleEndian) Endian endian = Endian.littleEndian;
     else                  Endian endian = Endian.bigEndian;
+    /++
+        CPU architecture.
+
+        Note:
+            This is intended for cases where you need a value representing the
+            CPU architecture at runtime. If you're doing something which should
+            compile differently depending on CPU architecture, then please use
+            `version (X86_64)`, `version (ARM)`, etc.
+
+        See_Also:
+            $(DDSUBLINK spec/version,PredefinedVersions, Predefined Versions)
+      +/
+    enum CPU
+    {
+        x86,   /// Intel and AMD 32-bit processors
+        x86_64, /// Intel and AMD 64-bit processors
+        arm, /// The ARM architecture (32-bit) (AArch32 et al)
+        aarch64, /// The Advanced RISC Machine architecture (64-bit)
+        asmJS, /// The asm.js intermediate programming language
+        avr, /// 8-bit Atmel AVR Microcontrollers
+        epiphany, /// The Epiphany architecture
+        ppc, /// The PowerPC architecture, 32-bit
+        ppc64, /// The PowerPC architecture, 64-bit
+        ia64, /// The Itanium architecture (64-bit)
+        mips32, /// The MIPS architecture, 32-bit
+        mips64, /// The MIPS architecture, 64-bit
+        msp430, /// The MSP430 architecture
+        nvptx, /// The Nvidia Parallel Thread Execution (PTX) architecture, 32-bit
+        nvptx64, /// The Nvidia Parallel Thread Execution (PTX) architecture, 64-bit
+        riscv32, /// The RISC-V architecture, 32-bit
+        riscv64, /// The RISC-V architecture, 64-bit
+        sparc, /// The SPARC architecture, 32-bit
+        sparc64, /// The SPARC architecture, 64-bit
+        s390, /// The System/390 architecture, 32-bit
+        systemZ, /// The System Z architecture, 64-bit
+        hppa, /// The HP PA-RISC architecture, 32-bit
+        hppa64, /// The HP PA-RISC architecture, 64-bit
+        sh, /// The SuperH architecture, 32-bit
+        webAssembly, /// The WebAssembly virtual ISA (instruction set architecture), 32-bit
+        alpha, /// The Alpha architecture
+        unknown, /// Unknown
+    }
+
+    /// The CPU architecture that the program was compiled for.
```

Comment:
In the age of Rosetta on MacOS I think this should be more explicit in the name

Reply:
It's just an emulator though? Can't be much different from what Qemu has been doing for at least decade and a bit.

Reply:
Some of these don't even have physical CPUs to begin with, like WebAssembly